client-v2/src/shared/util/insertAndDedupeSiblings.ts

```diff
@@ -29,7 +32,7 @@ export const insertAndDedupeSiblings = (
         // keep anything that doesn't match on id or is the item we just
         // inserted
         !insertionIDs.includes(siblingArticleFragment.id) ||
-        (isInsertionGroup && i === index)
+        (isInsertionGroup && isAnInsertedItem(i))
```

Comment:
Previously we were only checking for one drop article to ignore when deduping - the article that was dropped. However, when articles are "flattened" we can actually end up having more than one article being added in this position. So instead of checking `index` to see if we were the dropped _article_, we check index plus the amount of articles we've actually inserted to see if we were one of the dropped _articles_.

Reply:
Thanks for the thorough explanation!